platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,201 @@
+
+1. [Run Webpack](https://github.com/department-of-veterans-affairs/vets-website/blob/43ea0bdccd5e53886e3e38ea27d3f8e8e7bd9038/script/build.sh#L55)
+2. [Run the content build](https://github.com/department-of-veterans-affairs/vets-website/blob/43ea0bdccd5e53886e3e38ea27d3f8e8e7bd9038/script/build.sh#L61)
+
+Previously, Webpack was entangled in the content build, but no longer. Brooks is currently working on moving the content build into a [`content-build`](https://github.com/department-of-veterans-affairs/content-build/) repo to separate it from `vets-website`.
+
+##### Content Sources
+
+Static content currently comes from two sources:
+
+1. The Drupal CMS
+2. The [`vagov-content` repo](https://github.com/department-of-veterans-affairs/vagov-content/blob/master/pages/404.md)
+
+This design doc will focus on validating content from the Drupal CMS.
+
+##### Amazon S3
+
+The frontend of VA.gov is served as static files via Amazon S3. Since `vets-website` currently handles both application and content building, a failure in one blocks both from being deployed to S3 during a full deploy.
+
+###### Broken Link Checking
+
+The [`check-broken-link` middleware](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/check-broken-links/index.js) is a Metalsmith plugin that happens near the end of the Metalsmith static content build pipeline. It does the following:
+
+1. Loop through all of the HTML files in the Metalsmith pipeline
+2. Extract all of the broken `href`/`src` values using the `getBrokenLinks` helper
+3. Formats the result into useful console output
+4. Breaks the build on production, and logs the output on lower environments
+
+###### Accessibility Checking
+
+The accessibility checks only happen in CI ([the Integration step in Jenkins](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile#L81), which only happens during the full `vets-website` build.
+
+Before accessibility checks can be run, the sitemap is generated via the [`create-sitemap` Metalsmith plugin](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/create-sitemaps.js). Nightwatch iterates over all the paths found in the sitemap, and runs the [aXe accessibility checker](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/testing/e2e/nightwatch-commands/axeCheck.js) to look for violations.
+
+When violations are found, the build breaks and violations must be reported manually.
+
+The accessibility checks are skipped during a content-only deploy.
```

Comment:
All tests (except link checking) are skipped during content-only deploys by design - for speed of publishing. The link-checking runs as part of the build itself rather than as a separate step whereas unit tests run before before e2e and accessibility tests afterwards.

Reply:
:point_up: That context is probably good to add to the doc, @billfienberg .